### DISCUSSION TITLE
How do I handle V8 Module Errors?

### DISCUSSION THREAD

**Discussion by Cormanz** [caoccao/Javet#52] [Q&A] [answered]
Hi there, this might be a bit of a silly question, but I'm trying to setup logic in Javet that makes it so:
- I can load an ES6 module
- If the module errors plainly, I can log the error (fully, with stack trace) into the console
- If the module has an error in a promise, I can log the error (fully, with stack trace) into the console and precede with the script

I don't see any ways to capture that info from ES6 modules though.
`[com.caoccao.javet.interop.V8Runtime] Received promise reject callback with event 0 PromiseRejectWithNoHandler.`
This is all that gets logged. It seems I'm able to attach a catch listener to the result of `module.execute()`, but for some reason that errors too. Can you tell me how I can best do this? Thanks!

**Comment by caoccao** [comment, marked as answer]
No silly questions at all.

Please checkout test case `com.caoccao.javet.values.reference.TestV8ValuePromise`. You may register a callback like the following.

``` java
IJavetPromiseRejectCallback callback = (event, promise, value) -> { ... };
v8Runtime.setPromiseRejectCallback(callback);
```

The reason that you get that error message is Javet has a built-in `com.caoccao.javet.utils.JavetPromiseRejectCallback` which does nothing but logs a warning message.

Note: `setPromiseRejectCallback` is recommended to be used in V8 mode only because Node.js has its own [way](https://nodejs.org/docs/latest-v14.x/api/process.html#process_event_rejectionhandled) of handling that.

``` js
const unhandledRejections = new Map();
process.on('unhandledRejection', (reason, promise) => {
  unhandledRejections.set(promise, reason);
});
process.on('rejectionHandled', (promise) => {
  unhandledRejections.delete(promise);
});
```

Please let me know if that covers your need, or new features may be proposed.

**Reply by Cormanz** [reply]
Oh wow, I did _not_ know it was that easy. Thank you very much for both your support in answering my questions and maintaining this project!
